### PR TITLE
[RELAY-825] Add email address validation

### PR DESCRIPTION
**Pull request by AipengChen**
This PR adds a new field `email` to validate if an email address is valid using `go-validator` library

### CODE REVIEW

File: s12/protobuf/proto/validator.proto [hunk 1]
```diff
@@ -36,4 +36,6 @@ extend google.protobuf.FieldOptions {
   optional int64 repeated_len_gte = 65212;
   // collection size lesser than or equal to this value.
   optional int64 repeated_len_lte = 65213;
+  // Uses a Golang's mail.ParseAddress to determine a email address is a valid RFC 5322 address
```
Comment (vinovarkey) [inline s12/protobuf/proto/validator.proto:39]: We might want to update this to reflect the package we are using.

**Review by vinovarkey — CHANGES_REQUESTED** [review]

File: s12/protobuf/proto/validator.proto (hunk 1 above)
Comment (AipengChen) [reply s12/protobuf/proto/validator.proto]: Updated

**Review by vinovarkey — APPROVED** [review]
LGTM. Nice addition.

**Comment by sc-peter** [comment]
Thanks Aipeng and as discussed I made a few changes to align this with the new validator logic. It's great that we can have this used already. Please review the changes and let me know what you think.

**Comment by sc-peter** [comment]
Also tagging @rogchap for a quick sanity check. Taken from the WiP new validator, we plan on merging the email validator already. This is a relatively simple change and isolated from the string validators. It depends on the `govalidator` package though which might not make too much sense just for email validation, but we will have more validators that use that package later.

**Review by sc-peter — COMMENTED** [review]
I shouldn't approve now it includes some of my code as well but it looks good to me.

File: protobuf/protoc-gen-govalidator/example/example.proto [hunk 2]
```diff
@@ -12,8 +12,8 @@ message ExampleMessage {
   string id = 1 [(validator.uuid) = true];
   // bytes can also be parsed as UUID with support for gogo
   bytes user_id = 2 [(validator.uuid) = true];
-  // strings can validate against a regular expresion
-  string email = 3 [(validator.regex) = "[a-z0-9!#$%&'*+/=?^_{|}~-]+(?:\\.[a-z0-9!#$%&'*+/=?^_{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?"];
+  // should be a valid email address
+  string email = 3 [(validator.email) = {}];
```
Comment (vinovarkey) [inline protobuf/protoc-gen-govalidator/example/example.proto:16]: @sc-peter @AipengChen 

Is this the correct usage if email is optional? 
```suggestion
  string email = 3 [(validator.email) = { optional = true }];
```

Can we update this line to show the usage of optional?

**Review by vinovarkey — CHANGES_REQUESTED** [review]

**Review by vinovarkey — APPROVED** [review]
LGTM

**Pull request merged**